frontend/src/host_orchestrator/orchestrator/createsnapshotaction.go

```diff
@@ -53,7 +51,7 @@ func NewCreateSnapshotAction(opts CreateSnapshotActionOpts) *CreateSnapshotActio
 }
 
 func (a *CreateSnapshotAction) Run() (apiv1.Operation, error) {
-	if err := a.validateRequest(); err != nil {
+	if err := (CreateSnapshotRequestValidator{a.req}).Validate(); err != nil {
```

Comment:
This doesn't need to be a class at this call site. It can be a function like `ValidateCreateSnapshotRequest(a.req)`

Reply:
Agree! Done.